src/components/ReleaseService/ReleasePlan/TriggerRelease/TriggerReleaseForm.tsx

```diff
@@ -58,6 +62,7 @@ export const TriggerReleaseForm: React.FC<Props> = ({
           <ReleasePlanDropdown
             name="releasePlan"
             helpText="The release you want to release to the environments in your target workspace."
+            onReleasePlanChange={handleReleaseNameChange}
```

Comment:
As discussed, Once the application is removed from the URL parameters, you can remove the local useState. Instead, you can fetch the ReleasePlans directly within the TriggerReleaseForm component. Then, use the value from the Formik hook useField('releasePlan') to retrieve the selected release plan. Finally, pass the application from the chosen release plan to the Snapshot dropdown.

```suggestion
            releasePlans={<release-plans-fetched-from-useReleasePlans-hook>}
```

Reply:
Is this close to what you had in mind? -> https://github.com/openshift/hac-dev/pull/990/files#diff-5793ae169199d7a4684f5fa9c14443c7b812b0881493d91c3c85e41fed823deeR28-R38